src/main/java/com/denizenscript/depenizen/bukkit/objects/residence/ResidenceTag.java

```diff
@@ -108,44 +114,73 @@ public String toString() {
 
     @Override
     public ObjectTag getObjectAttribute(Attribute attribute) {
-        if (attribute == null) {
-            return null;
-        }
+        return tagProcessor.getObjectAttribute(this, attribute);
+    }
+
+    public static void registerTags() {
 
         // <--[tag]
         // @attribute <ResidenceTag.name>
         // @returns ElementTag
         // @plugin Depenizen, Residence
         // @description
-        // Returns the name of the residence.
+        // Returns the name of this Residence.
         // -->
-        if (attribute.startsWith("name")) {
-            return new ElementTag(residence.getName()).getObjectAttribute(attribute.fulfill(1));
-        }
+        tagProcessor.registerTag(ElementTag.class, "name", (attribute, object) -> {
+            return new ElementTag(object.getResidence().getName());
+        });
 
         // <--[tag]
         // @attribute <ResidenceTag.owner>
         // @returns PlayerTag
         // @plugin Depenizen, Residence
         // @description
-        // Returns the owner of the residence.
+        // Returns a PlayerTag of owner in this Residence.
```

Comment:
I think this was better as it was before? the return type is specified in the `@returns` meta